plugins/vlc-video/CMakeLists.txt

```diff
@@ -1,38 +1,43 @@
-project(vlc-video)
+cmake_minimum_required(VERSION 3.16...3.25)
+
+legacy_check()
+
+macro(check_vlc_path)
+  if(OS_WINDOWS OR OS_MACOS)
+    if(NOT EXISTS "${VLC_PATH}/include/vlc/libvlc.h")
+      message(FATAL_ERROR "Unable to find libvlc.h. Check if VLC_PATH set correctly.")
+    endif()
+
+    target_include_directories(vlc-video PRIVATE "${VLC_PATH}/include/vlc")
+  elseif(OS_LINUX OR OS_FREEBSD)
+    find_package(PkgConfig QUIET)
+    if(PKG_CONFIG_FOUND)
+      pkg_check_modules(PC_Libvlc QUIET libvlc)
+    endif()
+
+    if(PC_Libvlc_INCLUDEDIR)
+      target_include_directories(vlc-video PRIVATE "${PC_Libvlc_INCLUDEDIR}/vlc")
+    endif()
+  endif()
+endmacro()
 
-option(ENABLE_VLC "Build OBS with VLC plugin support" ${OS_LINUX})
+option(ENABLE_VLC "Build OBS with VLC plugin support" OFF)
 if(NOT ENABLE_VLC)
-  obs_status(DISABLED "vlc-video")
-  obs_status(
-    WARNING
-    "VLC plugin supported is not enabled by default - please switch ENABLE_VLC to ON to enable this functionality."
-  )
-  return()
+  target_disable(vlc-video)
```

Comment:
Just so that it doesn't get lost (I mentioned it off-thread already), the return statement seems to be missing here